beacon_node/http_api/src/lib.rs

```diff
@@ -3562,102 +3561,116 @@ pub fn serve<T: BeaconChainTypes>(
         .or(warp::post().and(
             post_beacon_blocks
-                .boxed()
-                .or(post_beacon_blinded_blocks.boxed())
-                .or(post_beacon_pool_attestations.boxed())
-                .or(post_beacon_pool_attester_slashings.boxed())
-                .or(post_beacon_pool_proposer_slashings.boxed())
-                .or(post_beacon_pool_voluntary_exits.boxed())
-                .or(post_beacon_pool_sync_committees.boxed())
-                .or(post_beacon_pool_bls_to_execution_changes.boxed())
-                .or(post_beacon_rewards_attestations.boxed())
-                .or(post_beacon_rewards_sync_committee.boxed())
-                .or(post_validator_duties_attester.boxed())
-                .or(post_validator_duties_sync.boxed())
-                .or(post_validator_aggregate_and_proofs.boxed())
-                .or(post_validator_contribution_and_proofs.boxed())
-                .or(post_validator_beacon_committee_subscriptions.boxed())
-                .or(post_validator_sync_committee_subscriptions.boxed())
-                .or(post_validator_prepare_beacon_proposer.boxed())
-                .or(post_validator_register_validator.boxed())
-                .or(post_lighthouse_liveness.boxed())
-                .or(post_lighthouse_database_reconstruct.boxed())
-                .or(post_lighthouse_database_historical_blocks.boxed())
-                .or(post_lighthouse_block_rewards.boxed())
-                .or(post_lighthouse_ui_validator_metrics.boxed())
-                .or(post_lighthouse_ui_validator_info.boxed())
+                .or(post_beacon_blinded_blocks)
+                .or(post_beacon_pool_attestations)
+                .or(post_beacon_pool_attester_slashings)
+                .or(post_beacon_pool_proposer_slashings)
+                .or(post_beacon_pool_voluntary_exits)
+                .or(post_beacon_pool_sync_committees)
+                .or(post_beacon_pool_bls_to_execution_changes)
+                .or(post_beacon_rewards_attestations)
+                .or(post_beacon_rewards_sync_committee)
+                .or(post_validator_duties_attester)
+                .or(post_validator_duties_sync)
+                .or(post_validator_aggregate_and_proofs)
+                .or(post_validator_contribution_and_proofs)
+                .or(post_validator_beacon_committee_subscriptions)
+                .or(post_validator_sync_committee_subscriptions)
+                .or(post_validator_prepare_beacon_proposer)
+                .or(post_validator_register_validator)
+                .or(post_lighthouse_liveness)
+                .or(post_lighthouse_database_reconstruct)
+                .or(post_lighthouse_database_historical_blocks)
+                .or(post_lighthouse_block_rewards)
+                .or(post_lighthouse_ui_validator_metrics)
+                .or(post_lighthouse_ui_validator_info)
                 .recover(warp_utils::reject::handle_rejection),
         ))
+        .boxed()
```

Comment:
Pushed a new commit that removes a bunch of the intermediate boxing. I think this box towards the end should do most of the heavy lifting by abstracting over all of the `and`s and `or`s for callers.

The weird thing is that the compiler seems to re-type-check this code when compiling futures, which I think is because it constructs a little state machine and jams all of the captured variables into it.